spinn_front_end_common/abstract_models/abstract_requires_tdma.py

```diff
@@ -0,0 +1,94 @@
+# Copyright (c) 2020-2021 The University of Manchester
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from six import add_metaclass
+from spinn_utilities.abstract_base import (
+    AbstractBase, abstractmethod, abstractproperty)
+
+
+@add_metaclass(AbstractBase)
+class AbstractRequiresTDMA(object):
```

Comment:
Why do you have this interface? It only has one direct implementation, and there doesn't appear to be any prospect of another one ever being created. Roll it into `RequiresTDMA` and don't proliferate interfaces that don't add value.

Reply:
so where ill agree given the removal via new type of app vertex. i feel ...... inclined to point out that there is indeed a second TDMA implementation floating around our tool chain. It just doesn't use any API's and assumes all vertices will use its generated agenda. it exists in fecs->interface_functions->tdma_agenda_builder. but that one ensures 1 packet on the ENTIRE network of interconnecting vertices. but noone has used it for an age

Reply:
done